Plugins/Covfie/src/FieldConversion.cpp

```diff
@@ -0,0 +1,208 @@
+
+  return {scaling * translation};
+}
+
+/// @brief Uses std::nextafter to generates a clamp backend
+/// configuration where arguments min and max hold floating point values.
+/// @param min (min_x, min_y, min_z)
+/// @param max (max_x, max_y, max_z)
+/// @return The clamp field configuration.
+template <typename backend_t, typename point3_1_t, typename point3_2_t>
+typename backend_t::configuration_t clampConfigurationFloat(
+    const point3_1_t& min, const point3_2_t& max) {
+  return {{std::nextafter(static_cast<float>(min[0]),
+                          std::numeric_limits<float>::infinity()),
+           std::nextafter(static_cast<float>(min[1]),
+                          std::numeric_limits<float>::infinity()),
+           std::nextafter(static_cast<float>(min[2]),
+                          std::numeric_limits<float>::infinity())},
+          {std::nextafter(static_cast<float>(max[0]),
+                          -std::numeric_limits<float>::infinity()),
+           std::nextafter(static_cast<float>(max[1]),
+                          -std::numeric_limits<float>::infinity()),
+           std::nextafter(static_cast<float>(max[2]),
+                          -std::numeric_limits<float>::infinity())}};
+}
+
+/// @brief Creates a covfie field from a generic magnetic field.
+/// @param magneticField The generic magnetic field.
+/// @param cache The cache.
+/// @param nBins 3D array of containing the number of bins for each axis.
+/// @param min (min_x, min_y, min_z)
+/// @param max (max_x, max_y, max_z)
+/// @return A clamp affine linear strided covfie field.
+template <typename magnetic_field_t, typename point3_1_t, typename point3_2_t,
+          typename point3_3_t>
+InterpolatedField covfieFieldLinear(const magnetic_field_t& magneticField,
+                                    typename magnetic_field_t::Cache& cache,
+                                    const point3_1_t& nBins,
+                                    const point3_2_t& min,
+                                    const point3_3_t& max) {
+  auto builder = newBuilder(magneticField, cache, nBins, min, max);
+  InterpolatedField field(covfie::make_parameter_pack(
+      clampConfigurationFloat<InterpolatedField::backend_t>(min, max),
+      affineConfiguration<InterpolatedField::backend_t::backend_t>(nBins, min,
+                                                                   max),
+      InterpolatedField::backend_t::backend_t::backend_t::configuration_t{},
+      builder.backend()));
+
+  return field;
+}
+
+/// @brief Creates a covfie field from a magnetic field provider by sampling it.
+/// @param magneticField The acts magnetic field provider.
+/// @param cache The acts cache.
+/// @param nBins 3D array of containing the number of bins for each axis.
+/// @param min (min_x, min_y, min_z)
+/// @param max (max_x, max_y, max_z)
+/// @return A clamp affine linear strided covfie field.
+InterpolatedField covfieField(const Acts::MagneticFieldProvider& magneticField,
```

Comment:
Just to not: if you have an `InterpolatedMagneticField` that's held as a `MagneticFieldProvider`, I believe it will go through this overload. Should we test this?

Reply:
I have added that the magnetic field provider must be defined within min (inclusive) and max (inclusive) as a comment for the function. For the covfieField function that takes an interpolated acts field, since you don't pass nBins, min, and max, hopefully, it's clear which of the two functions you are using.